src/main/java/dev/morling/onebrc/CalculateAverage_abhinavupadhyay.java

```diff
@@ -72,7 +71,21 @@ public void put(long cityStartOffset, long cityLength, long nameHash, int temper
                 for (; i < cityLength; i++) {
                     array[i] = UNSAFE.getByte(cityStartOffset++);
                 }
-                table[index] = new Row(new String(array, 0, i), temperature, temperature, 1, temperature, (int) uhash);
+                table[index] = new Row(new String(array, 0, i), temperature, temperature, 1, temperature, uhash);
+                return;
+            }
+
+            while (row.hash != uhash) {
```

Comment:
Just checking the hash isn't enough, you'll need to compare the actual name to differentiate between multiple stations with the same hash.